tests/helpers/Participant.ts

```diff
@@ -0,0 +1,238 @@
+    };
+
+    /**
+     * Creates a participant with given name.
+     * @param name
+     */
+    constructor(name: string) {
+        this._name = name;
+    }
+
+    /**
+     * Returns participant endpoint ID.
+     * @returns {Promise<string>} The endpoint ID.
+     */
+    async getEndpointId() {
+        if (!this._endpointId) {
+            this._endpointId = await this.driver.execute(() => { // eslint-disable-line arrow-body-style
+                return APP.conference.getMyUserId();
+            });
+        }
+
+        return this._endpointId;
+    }
+
+    /**
+     * The driver it uses.
+     */
+    get driver() {
+        return browser.getInstance(this._name);
+    }
+
+    /**
+     * The name.
+     */
+    get name() {
+        return this._name;
+    }
+
+    /**
+     * Joins conference.
+     */
+    async joinConference(context: { roomName: string }, skipInMeetingChecks = false) {
+        this.context = context;
+
+        const url = urlObjectToString({
+            room: context.roomName,
+            configOverwrite: this.config,
+            interfaceConfigOverwrite: {
+                SHOW_CHROME_EXTENSION_BANNER: false
+            },
+            userInfo: {
+                displayName: this._name
+            }
+        });
+
+        await this.driver.setTimeout({ 'pageLoad': 30000 });
+
+        await this.driver.url(url);
+
+        await this.waitForPageToLoad();
+
+        await this.waitToJoinMUC();
+
+        await this.postLoadProcess(skipInMeetingChecks);
+    }
+
+    /**
+     * Loads stuff after the page loads.
+     * @private
+     */
+    private async postLoadProcess(skipInMeetingChecks) {
+        const driver = this.driver;
+
+        const parallel = [];
+
+        parallel.push(driver.execute((name, sessionId) => {
+            APP.UI.dockToolbar(true);
+
+            // disable keyframe animations (.fadeIn and .fadeOut classes)
+            $('<style>.notransition * { '
+                + 'animation-duration: 0s !important; -webkit-animation-duration: 0s !important; transition:none; '
+                + '} </style>').appendTo(document.head);
+
+            $('body').toggleClass('notransition');
+
+            document.title = `${name}`;
+
+            console.log(`lib-jitsi-meet version: ${JitsiMeetJS.version} sessionId: ${sessionId}`);
+        }, this._name, driver.sessionId));
+
+        // disable the blur effect in firefox as it has some performance issues
+        if (driver.capabilities.firefox) {
+            parallel.push(driver.execute(() => {
+                try {
+                    const blur = document.querySelector('.video_blurred_container');
+
+                    if (blur) {
+                        document.querySelector('.video_blurred_container').style.display = 'none';
+                    }
+                } catch (e) {} // eslint-disable-line no-empty
+            }));
+        }
+
+        if (skipInMeetingChecks) {
+            await Promise.all(parallel);
```

Comment:
all will fail on the first failure, do you want that or allSettled, which waits for all of them to finish?

Reply:
Actually this change is no good, as it skips some rejection and failures.